meta/store.go

```diff
@@ -994,11 +1008,28 @@ func (s *Store) Authenticate(username, password string) (ui *UserInfo, err error
 			return ErrUserNotFound
 		}
 
+		// Check the local auth cache first.
+		s.mu.RLock()
+		p, ok := s.authCache[username]
+		s.mu.RUnlock()
```

Comment:
Not sure either of these options will work.  If the password was altered, then the value we'd be adding in the cache is still the old one because we read it before the update.  The user would not be in the cache at this point.  Even it was, we'd still be setting the wrong value again.